### PR TITLE
Add support for multiple keys

### PR DESCRIPTION
**Pull request by DanielCKennedy**
Adds support for a single encryption-provider pod to handle multiple keys listening on different unix sockets. 

This is useful when switching keys for encryption or if long-term support of multiple keys is needed.

### CODE REVIEW

**Comment by DanielCKennedy** [comment]
Questions:

1. What's the best way to support inputs for multiple keys?

(a) comma-separated list of keys + sockets overloading existiing `key` and `listen` fields. Match key and sockets based on index in list
```
--key key1,key2
--listen socket1,socket2 
```
(b) multiple key + socket input fields. Match key and sockets based on index
```
--key key1
--listen socket1
--key key2
--listen socket2
```
(b) Structured config input that explicitly groups keys to sockets:
```
--key-configs {key=key1,listen=socket1},{key=key2,listen=socket2}
```

2. How should we handle the healthz endpoint?

Currently we just use v1 plugin healthz. With multiple keys should we include both keys in the healthz check? Should we include both v1 and v2 plugins healthcheck for each key?

**Comment by DanielCKennedy** [comment]
We currently only uses the v1 health check of the first key. This will mess up metrics to do v1 encrypts for keys that are meant to only use v2. I added logic to health check endpoint to check each key using both the v1 and v2 plugins that's configurable via a confusing list of kmsVersions. I think I'm going to change it to only do either v1 or v2 for both keys

File: cmd/server/main.go [hunk 1]
```diff
@@ -48,6 +49,11 @@ func main() {
 	)
 	flag.Parse()
 
+	if len(*keys) != len(*addrs) {
```
Comment (wongma7) [inline cmd/server/main.go:52]: now key is a required argument where before it was not, this is technically a breaking change. what happened before if key was left empty, would key-server just idle?

File: pkg/healthz/healthz.go [hunk 2]
```diff
@@ -10,21 +10,33 @@ import (
...
-		rw.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprint(rw, err)
-		zap.L().Error("health check failed", zap.Error(err))
-		return
+	for _, p := range hd.p1s {
+		err := p.Health()
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprint(rw, err)
+			zap.L().Error("health check failed", zap.Error(err))
```
Comment (wongma7) [inline pkg/healthz/healthz.go:28]: will the naked err will tell us which plugin failed? if not please include the keyId in the error, like health check for plugin p.keyId failed or something

Comment (wongma7) [reply pkg/healthz/healthz.go:28]: Better yet we could poll each one and record the health of each one and emit an aggregate error.

but if you don't think this is going to be of any use in practice then ignore. I think in general we do want to know which of the 2+ plugins failed though.

File: cmd/server/main.go (hunk 1 above)
Comment (DanielCKennedy) [reply cmd/server/main.go:52]: it should still work since we set a default value of [""]. len(keys) == 1

File: pkg/healthz/healthz.go (hunk 2 above)
Comment (DanielCKennedy) [reply pkg/healthz/healthz.go:28]: This will need more thought. We'll follow up in another PR if this proves to make debugging more difficult.

**Pull request merged**